src/main/scala/org/hammerlab/guacamole/commands/VariantSupport.scala

```diff
@@ -110,23 +109,26 @@ object VariantSupport {
         ).reduce(_ ++ _)
 
       alleleCounts.saveAsTextFile(args.output)
-
     }
 
     /**
      * Count alleles in a pileup
      *
-     * @param pileup Pileup of reads a given locu
-     * @return Iterator of AlleleCount which contains pair of reference and alternate with a count
+     * @param pileup Pileup of reads a given locus.
+     * @return Iterator of AlleleCount which contains pair of reference and alternate with a count.
      */
-    def pileupToAlleleCounts(pileup: Pileup): Iterator[AlleleCount] = {
-      val alleles = pileup.elements.groupBy(_.allele)
-      alleles.map(kv => AlleleCount(pileup.sampleName,
-        pileup.contigName,
-        pileup.locus,
-        Bases.basesToString(kv._1.refBases),
-        Bases.basesToString(kv._1.altBases),
-        kv._2.size)).iterator
-    }
+    def pileupToAlleleCounts(pileup: Pileup): Iterator[AlleleCount] =
+      (for {
+        (allele, elements) <- pileup.elements.groupBy(_.allele)
```

Comment:
`for`s generally allow writing a lot less cruft, so I prefer them; when you want a destructuring assignment as well, they are even better.